scripts/basic-workflow-tests.sh

```diff
@@ -170,13 +198,15 @@ test_serve() {
 }
 
 test_chat() {
+    local served_model
     task Chat with the model
-    CHAT_ARGS=()
-    if [ "$GRANITE" -eq 1 ]; then
-        CHAT_ARGS+=("-m" "${DATA_HOME}/instructlab/models/granite-7b-lab-Q4_K_M.gguf")
-    elif [ "$MIXTRAL" -eq 1 ]; then
-        CHAT_ARGS+=("-m" "${DATA_HOME}/instructlab/models/mixtral-8x7b-instruct-v0.1.Q4_K_M.gguf" "--model-family" "mixtral")
+
+    served_model=$(get_served_model_from_server)
+    CHAT_ARGS=('-m' "${served_model}")
```

Comment:
It's not the same under-the-hood. The current default for `ilab chat` if the model isn't specified is `DEFAULTS.DEFAULT_MODEL` or what's specified in the user's config. So if we're serving a different model then it'll fail to find it.

Reply:
This is not how it works, please look at: https://github.com/instructlab/instructlab/blob/503d0d3bdf7a40966de7ac94f6b93903628870d2/src/instructlab/model/chat.py#L208-L241.

The logic is that if flag/config value for the model_path are the same and we find a different version on the server, then we use the model reported by the server, that's why we don't need to `-m` normally.

Unfortunately, this code is broken, mainly because of https://github.com/instructlab/instructlab/issues/1736 which I'm currently fixing.

Reply:
This makes sense, let's merge your PR #1754 and then we can rollback this change.